### ISSUE TITLE
music21.converter.parse() raises an ABCHandlerException when given an abc from the specification

### ISSUE THREAD

**Issue by jay-tyler** [cuthbertLab/music21#178]
I'm sort of new to music21 and thought I would attempt to import an abc file to get a sense of how everything works.

I'm using the Reels.abc example from section 13.3 of the abc specification here:
http://abcnotation.com/wiki/abc:standard:v2.1#abc_tune_definition

I've validated that the file parses appropriately using EasyABC, but using music21 result in an ABCHandlerException.

Here's the full traceback:
```
In [17]: abcScore = m.converter.parse('reels.abc')
---------------------------------------------------------------------------
ABCHandlerException                       Traceback (most recent call last)
<ipython-input-17-d2687cd83db9> in <module>()
----> 1 abcScore = m.converter.parse('reels.abc')

/Users/jtyler/virtualenvs/music21/lib/python2.7/site-packages/music21/converter/__init__.pyc in parse(value, *args, **keywords)
   1067         return parseData(value, number=number, format=m21Format, **keywords)
   1068     elif os.path.exists(value):
-> 1069         return parseFile(value, number=number, format=m21Format, forceSource=forceSource, **keywords)
   1070     elif (valueStr.startswith('http://') or valueStr.startswith('https://')):
   1071         # its a url; may need to broaden these criteria

/Users/jtyler/virtualenvs/music21/lib/python2.7/site-packages/music21/converter/__init__.pyc in parseFile(fp, number, format, forceSource, **keywords)
    962     '''
    963     v = Converter()
--> 964     v.parseFile(fp, number=number, format=format, forceSource=forceSource, **keywords)
    965     return v.stream
    966

/Users/jtyler/virtualenvs/music21/lib/python2.7/site-packages/music21/converter/__init__.pyc in parseFile(self, fp, number, format, forceSource, storePickle, **keywords)
    498         else:
    499             environLocal.printDebug("Loading original version")
--> 500             self.parseFileNoPickle(fp, number, format, forceSource)
    501             if writePickle is True and fpPickle is not None and storePickle is True:
    502                 # save the stream to disk...

/Users/jtyler/virtualenvs/music21/lib/python2.7/site-packages/music21/converter/__init__.pyc in parseFileNoPickle(self, fp, number, format, forceSource, **keywords)
    439         self.setSubconverterFromFormat(useFormat)
    440         self.subConverter.keywords = keywords
--> 441         self.subConverter.parseFile(fp, number=number)
    442         self.stream.filePath = fp
    443         self.stream.fileNumber = number

/Users/jtyler/virtualenvs/music21/lib/python2.7/site-packages/music21/converter/subConverters.pyc in parseFile(self, fp, number)
    952             # is given
    953             self.stream = abcFormat.translate.abcToStreamOpus(abcHandler,
--> 954                            number=number)
    955         # just get a single work
    956         else:

/Users/jtyler/virtualenvs/music21/lib/python2.7/site-packages/music21/abcFormat/translate.pyc in abcToStreamOpus(abcHandler, inputM21, number)
    406     # returns a dictionary of numerical key
    407     if abcHandler.definesReferenceNumbers():
--> 408         abcDict = abcHandler.splitByReferenceNumber()
    409         if number != None and number in abcDict:
    410             # get number from dictionary; set to new score

/Users/jtyler/virtualenvs/music21/lib/python2.7/site-packages/music21/abcFormat/__init__.pyc in splitByReferenceNumber(self)
   2245         #environLocal.printDebug(['pairs:', pairs, 'keys', keys])
   2246         if len(keys) != len(pairs):
-> 2247             raise ABCHandlerException('cannot match pairs to keys: %s, %s' % (pairs, keys))
   2248         # case of one reference number:
   2249         if len(pairs) == 1: # one tune defined

ABCHandlerException: cannot match pairs to keys: [[0, 3], [3, 103], [103, 215]], [1, 2]
```

**Comment by jay-tyler** [comment]
Also - I tried opening this same file using abcFormat.ABCFile.open("foo.abc")

The method signature and online documentation suggest that this particular method takes a filename, but it actually seems to expect an ABCFile instance instead. It seems either the method or documentation need to be updated.

Should I open a separate issue for this?

The full traceback is here:
```
In [22]: m.abcFormat.ABCFile.open("foo.abc")
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-22-994bf596261f> in <module>()
----> 1 m.abcFormat.ABCFile.open("foo.abc")

TypeError: unbound method open() must be called with ABCFile instance as first argument (got str instance instead)
```

**Comment by mscuthbert** [comment]
converter.parse() is definitely the correct thing to call.  Everything else is very advanced.  I'll look at the reels and see what's up

**Comment by mscuthbert** [comment]
problem confirmed.

**Comment by mscuthbert** [comment]
Fixed in 0bcc079. Problem was that we were not implementing the ABC 2.0 (2003) File Header standard, where metadata to be applied to each song of the file can live before the first X: tag.  This should work now and lots more tests along the way.

for anyone else who sees this: ABC 2.0 is the latest version music21 aims to support for now -- so 2.1 specific examples would be feature requests.  but definitely bugs we want to squash.  Thanks @jay-tyler

**Comment by jay-tyler** [comment]
Thanks for the quick fix @mscuthbert!

Just a got a bleeding-edge install via `pip install git+https://github.com/cuthbertLab/music21.git`
